src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs

```diff
@@ -393,6 +393,46 @@ private static IncrementalMethodStubGenerationContext CalculateStubInformation(M
             // Create the stub.
             var signatureContext = SignatureContext.Create(symbol, DefaultMarshallingInfoParser.Create(environment, generatorDiagnostics, symbol, new InteropAttributeCompilationData(), generatedComAttribute), environment, typeof(VtableIndexStubGenerator).Assembly);
 
+            for (var i = 0; i < signatureContext.ElementTypeInformation.Length; ++i)
+            {
+                if (signatureContext.ElementTypeInformation[i].IsManagedReturnPosition)
+                {
+                    if (signatureContext.ElementTypeInformation[i].ManagedType == SpecialTypeInfo.Void)
+                    {
+                        signatureContext = signatureContext with
+                        {
+                            ElementTypeInformation = signatureContext.ElementTypeInformation.RemoveAt(i)
+                        };
+                    }
+                    else
+                    {
+                        var managedSignatureAsNativeOut = signatureContext.ElementTypeInformation[i] with
+                        {
+                            RefKind = RefKind.Out,
+                            RefKindSyntax = SyntaxKind.OutKeyword,
+                            ManagedIndex = TypePositionInfo.ReturnIndex,
+                            NativeIndex = symbol.Parameters.Length
+                        };
+                        signatureContext = signatureContext with
+                        {
+                            ElementTypeInformation = signatureContext
+                                .ElementTypeInformation
+                                .SetItem(i, managedSignatureAsNativeOut)
+                        };
+                    }
+                    break;
+                }
+            }
+
+            signatureContext = signatureContext with
```

Comment:
Why? I don't see how this is all connected above. We are looping and then always creating another `signatureContext`? I'm not saying this is wrong but we need to comment this code. None of it is "obvious".